src/pip/_internal/network/auth.py

```diff
@@ -152,9 +152,8 @@ def _get_new_credentials(self, original_url, allow_netrc=True,
         # If we don't have a password and keyring is available, use it.
         if allow_keyring:
             # The index url is more specific than the netloc, so try it first
-            kr_auth = (
-                get_keyring_auth(index_url, username) or
-                get_keyring_auth(netloc, username)
+            kr_auth = get_keyring_auth(index_url, username) or get_keyring_auth(
+                netloc, username
             )
```

Comment:
That's what enforcing line lengths over readability gets you (this is one place I *like* black, it allows a little leeway before enforcing the line length - just not this much).

IMO this just needs rewriting. Personally I'd shorten some names so it can be written on one line, the alternative being

```python
kr_auth = get_keyring_auth(index_url, username)
if not kr_auth:
    kr_auth = get_keyring_auth(netloc, username)
```

or

```python
kr_auth = (
    get_keyring_auth(index_url, username) or
    get_keyring_auth(netloc, username)
)
```

Black would probably reformat that last one too, though 🙂